Add a C++ quick-start to the viewer (#3961)
rerun-io/rerun: crates/re_viewer/data/quick_start_guides/cpp_connect.md

## C++ Quick Start

${SAFARI_WARNING}

#### Installing the Rerun viewer
The Rerun C++ SDK works by connecting to an awaiting Rerun Viewer over TCP.

Building from source:
* [Install cargo](https://rustup.rs/)
* `cargo install rerun-cli`

Using `pip`:
* `pip install rerun-sdk`

After you've installed it, type `rerun` in your terminal to start the viewer.

#### Using the Rerun C++ SDK

> First install the `arrow-cpp` library on your system using your favorite package manager.

Reviewer: I'd suggest we add something like

```
Find more information at the official Arrow Apache [install guide](https://arrow.apache.org/install/).
```

I'm also leaning towards mentioning `pixi global install arrow-cpp` because that seems to be one of the easiest ways and I think we should start advertising it. But I don't know yet how to formulate this in the least awkward way.

> Then add the following to your `CMakeLists.txt`:

Reviewer: ```suggestion
If you're using CMake, you can add the following to your `CMakeLists.txt` to add a rerun_cpp target to your application. 
```
...

```
This download a bundle with pre-built Rerun C static libraries for most desktop platforms, all Rerun C++ sources and headers, as well as CMake build instructions for them.
```

Reviewer: Thinking about it a bit more, maybe that's a separate section somewhere that documents how the Rerun C++ sdk is built up

```cmake
include(FetchContent)

> FetchContent_Declare(rerun_sdk URL https://build.rerun.io/commit/06dd483/rerun_cpp_sdk.zip) # 2023-10-20

Reviewer 2: ```suggestion
FetchContent_Declare(rerun_sdk URL https://github.com/rerun-io/rerun/latest/download/rerun_cpp_sdk.zip)
```

Author: 404

Reviewer 2: Yeah it will 404 until we have a cpp release

Reviewer 2: I've temporarily added the C++ sdk as an asset of the 0.9.1 release so we can test our links.

Reviewer: 'latest' is actually a bad idea for a users's cmake script. Because then it auto-updates when we bring out a new latest

Reviewer 2: `https://github.com/rerun-io/rerun/releases/download/0.10.0/rerun_cpp_sdk.zip` then

FetchContent_MakeAvailable(rerun_sdk)

> ```

Reviewer: I'll document how to build from source as well in a follow-up PR!

> Make sure you link with `rerun_sdk`.

Reviewer: ```suggestion
Make sure you link with `rerun_sdk`:
```cmake
target_link_libraries(your_executable PRIVATE rerun_sdk)
```
```
